kong/plugins/redirect/schema.lua

```diff
@@ -0,0 +1,44 @@
+local typedefs = require "kong.db.schema.typedefs"
+
+return {
+    name = "redirect",
+    fields = {
+        {
+            protocols = typedefs.protocols_http
+        },
+        {
+            config = {
+                type = "record",
+                fields = {
+                    {
+                        status_code = {
+                            description = "The response code to send. Must be an integer between 100 and 599.",
+                            type = "integer",
+                            required = true,
+                            default = 301,
+                            between = { 100, 599 }
+                        }
+                    },
+                    {
+                        -- This is intentionally flexible and does not require a http/https prefix in order to support
+                        -- redirecting to uris such as someapp://path
+                        location = {
+                            description = "The URL to redirect to",
+                            type = "string",
+                            required = true
+                        }
```

Comment:
I believe the default url type admits prefixes other than http/https while still providing some basic url validation
```suggestion
                        location = typedefs.url
```